MM.lua

```diff
@@ -87,3 +87,9 @@ function MM:updateGradInput(input, gradOutput)
 
   return self.gradInput
 end
+
+function MM:clearState()
+   local retval = parent.clearState(self)
+   self.gradInput = {torch.Tensor(), torch.Tensor()} -- re-initialize
```

Comment:
thanks! torch[self._type] didn't work because self._type was "torch.FloatTensor" or "torch.CudaTensor", when it needed to be "FloatTensor" or "CudaTensor"

in the end, I went a different way